### PR TITLE
DB-3192: Bubble Up Surrogate Keys On Outgoing Responses

### PR DESCRIPTION
**Pull request by CobyPear**
## What changes were made?
- [cms-kit] Move addSurrogateKeyHeader to new cms-kit package
- [wp-kit] Add pantheon-debug header to wp-kit graphql client outgoing requests
- [next-wp] Refactor data fetchers to return headers
- [next-wp] Combine headers and add them to res in getServerSideProps using addSurrogateKeyHeader
- [next-wp] Add helper to get unique keys from multiple surrogate-key-raw headers

## Where were the changes made?
new cms-kit, wp-kit, drupal-kit, and next-wp-starter

## How have the changes been tested?
Tests written as needed
## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

**Review by backlineint — CHANGES_REQUESTED** [review]
Working well - just a few minor things and questions. In addition to what is included inline:

* For a site that does not have the pantheon-advanced-page-cache activated, the Surrogate-Key-Raw header is set, but it is empty. In that case, I'd expect the header to not be set at all.

File: packages/wordpress-kit/index.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { gql } from 'graphql-request';
+import { addSurrogateKeyHeader } from '@pantheon-systems/cms-kit';
 
-export { gql };
+export { gql, addSurrogateKeyHeader };
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply packages/wordpress-kit/index.ts:4]: No good reason, it should be exported from both :) Change incoming.

File: starters/next-wordpress-starter/pages/index.jsx [hunk 2]
```diff
@@ -53,10 +57,12 @@ export default function Home({ menuItems, posts }) {
...
 export async function getServerSideProps({ res }) {
-	const menuItems = await getFooterMenu();
-	const posts = await getLatestPosts(12);
-	setEdgeHeader({ res });
+	const { menuItems, menuItemHeaders } = await getFooterMenu();
+	const { posts, headers: postHeaders } = await getLatestPosts(12);
 
+	const keys = getSurrogateKeys({ headers: [menuItemHeaders, postHeaders] });
+	addSurrogateKeyHeader(keys, res);
+	setEdgeHeader({ res });
```
[the remark this answers is not in the document]
Comment (CobyPear) [reply starters/next-wordpress-starter/pages/index.jsx:65]: Yeah that makes sense to me. So I would essentially be combining these operations into one function?

Related, I propose we decide on "add" or "set" and refactor accordingly. I'm not a huge fan of using both as it's a bit confuisng. Which do you think we should use?

[the remark this answers is not in the document]
Comment (backlineint) [reply starters/next-wordpress-starter/pages/index.jsx]: Saw that you landed on set - I think that makes sense.

**Review by backlineint — APPROVED** [review]

**Pull request merged**